src/gui/uithememanager.cpp

```diff
@@ -140,5 +156,41 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
         return path;
     }
 #endif
-    return IconProvider::instance()->getIconPath(iconId);
+
+    const auto iter = m_iconMap.find(iconId);
+    QString iconPath = iter != m_iconMap.end() ? iter->toString() : QString{};
+    QString iconIdPattern = iconId;
+    while (iconPath.isEmpty() && !iconIdPattern.isEmpty()) {
+        const int sepIndex = iconIdPattern.indexOf('.');
+        iconIdPattern = "*" + (sepIndex == -1 ? "" : iconIdPattern.right(iconIdPattern.size() - sepIndex));
+        const auto iter = m_iconMap.find(iconIdPattern);
+        if (iter != m_iconMap.end())
+            iconPath = iter->toString();
```

Comment:
You should check for empty icon path here since it is error case.
Also you should cache icon path for given icon id in m_iconMap to avoid searching by pattern next time.